usecases/modules/modules.go

```diff
@@ -247,6 +247,24 @@ func (p *Provider) isOnlyOneModuleEnabledOfAGivenType(moduleType modulecapabilit
 	return i == 1
 }
 
+func (p *Provider) IsGenerative(modName string) bool {
+	for _, mod := range p.registered {
+		if mod.Name() == modName {
+			return mod.Type() == modulecapabilities.Text2TextGenerative
+		}
+	}
+	return false
+}
+
+func (p *Provider) IsReranker(modName string) bool {
+	for _, mod := range p.registered {
```

Comment:
please use `p.GetByName()` method as it takes modules alternative names into account